### PR TITLE
Simplify ResourceCleaner a bit, add a bunch of comments/tests

### PR DESCRIPTION
**Pull request by nickbp**

### CODE REVIEW

File: src/main/java/org/apache/mesos/offer/ResourceCleaner.java [hunk 1]
```diff
@@ -2,173 +2,203 @@
...
-      }
+    /**
+     * Creates a new {@link ResourceCleaner} using the provided expected resource information.
+     *
+     * @param expectedResources a list of all expected resources
+     */
+    public ResourceCleaner(Collection<Resource> expectedResources) {
+        this.expectedPersistentVolumeIds = getPersistentVolumeIds(expectedResources);
+        this.expectedReservedResourceIds = getReservedResourceIds(expectedResources);
+        // TODO(nick) TEMP
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/offer/ResourceCleaner.java]: I don't understand this TODO statement.

File: src/main/java/org/apache/mesos/offer/ResourceCleaner.java [hunk 2]
```diff
@@ -2,173 +2,203 @@
...
+        for (Offer offer : offers) {
+            offerResourceCount += offer.getResourcesCount();
+            for (Resource toDestroy : selectUnexpectedResources(
+                    expectedPersistentVolumeIds, getPersistentVolumesById(offer))) {
+                recommendations.add(new DestroyOfferRecommendation(offer, toDestroy));
+            }
+        }
+        int destroyRecommendationCount = recommendations.size();
+
+        // Then, find any unexpected persistent volumes ANDresource reservations which should (both)
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/offer/ResourceCleaner.java]: need a `space` to fix `ANDresource`

File: src/main/java/org/apache/mesos/offer/ResourceCleaner.java (hunk 1 above)
Comment (nickbp) [inline src/main/java/org/apache/mesos/offer/ResourceCleaner.java]: Whoops, I left that for myself to clean up later. Fixed now!

File: src/main/java/org/apache/mesos/offer/ResourceCleaner.java (hunk 2 above)
Comment (nickbp) [inline src/main/java/org/apache/mesos/offer/ResourceCleaner.java]: Just saw that myself, fixed now

File: src/test/java/org/apache/mesos/offer/ResourceCleanerTest.java [hunk 3]
```diff
@@ -10,138 +11,266 @@
...
+        when(mockStateStore.fetchTasks("b"))
+                .thenReturn(Arrays.asList(TASK_INFO_2));
+        populatedCleaners.add(new ResourceCleaner(mockStateStore));
+
+        allCleaners.addAll(emptyCleaners);
+        allCleaners.addAll(populatedCleaners);
+    }
+
+    @Test
+    public void testNoOffers() {
```
Comment (gabrielhartmann) [inline src/test/java/org/apache/mesos/offer/ResourceCleanerTest.java]: Could you add another test which generates no recommendations when presented with All expected resources and volumes?

Comment (nickbp) [inline src/test/java/org/apache/mesos/offer/ResourceCleanerTest.java]: Done ("testPopulatedCleanersAllExpected")

**Pull request merged**